tt_metal/common/test_tiles.hpp

```diff
@@ -11,6 +11,7 @@
 #include <cstdint>
 #include <vector>
 #include <optional>
+#include <span>
```

Comment:
We have `"tt_metal/tt_stl/span.hpp"` which has `tt::stl::Span<T>` instead of `std::span<T>`. I understand that ttnn only compiles with C++20 and it's unlikely that will change, but this would help bring some consistency within the codebase, and the `tt::stl::Span<T>` supports [the constructor from `std::initializer_list<T>`](https://en.cppreference.com/w/cpp/container/span/span) introduced in C++26, which was very useful at least for tt_metal.

Reply:
This would also allow a lot of the places you currently use `SmallVector` parameters to accept what used to be initializer lists to go back to using initializer lists.

Reply:
Yep, will do. 
We should probably update `best_practices.md` which says to use `std::span` cc @ayerofieiev-tt 

@patrickroberts I think it's a bit odd that `"tt_metal/tt_stl/span.hpp"` actually requires both C++20 and boost to compile (`explicit(bool)` is C++ 20), probably we should address it.

Reply:
I updated all of the usages in ttnn to use `tt::stl::Span` and updated `best_practices.md`

Reply:
@sminakov-tt `explicit(bool)` was an oversight, that was not an intentional use of C++20, and will be addressed when we get time to work towards C++17 compatibility in tt-metal. If you see anything else in that header which requires C++20 please feel free to point it out because it shouldn't be there.